From ad68413e363e51af7259d0fd3c69d5ba2974f590 Mon Sep 17 00:00:00 2001
From: Sebastian Bazley <sebb@apache.org>
Date: Fri, 15 Aug 2008 18:16:59 +0000
Subject: [PATCH] NET-230 - ParserInitializationException when connecting to a
 Unix FTP server: comparison string must be upper case. Also added check for
 AS/4000 == OS/400

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/net/trunk@686304 13f79535-47bb-0310-9956-ffa450edef68
---
 src/java/org/apache/commons/net/ftp/FTPClientConfig.java | 8 +++++++-
 .../net/ftp/parser/DefaultFTPFileEntryParserFactory.java | 5 +++--
 .../ftp/parser/DefaultFTPFileEntryParserFactoryTest.java | 9 +++++++++
 xdocs/changes.xml                                        | 4 ++++
 4 files changed, 23 insertions(+), 3 deletions(-)

diff --git a/src/java/org/apache/commons/net/ftp/FTPClientConfig.java b/src/java/org/apache/commons/net/ftp/FTPClientConfig.java
index 05b3a85d2..1239bcb08 100644
--- a/src/java/org/apache/commons/net/ftp/FTPClientConfig.java
+++ b/src/java/org/apache/commons/net/ftp/FTPClientConfig.java
@@ -171,6 +171,12 @@ public class FTPClientConfig
      */
     public static final String SYST_OS400 = "OS/400";
     
+
+    /**
+     * Alternate SYST value for an AS/400 system.
+     */
+    public static final String SYST_AS400 = "AS/400";
+
     /**
      * Identifier by which an MVS-based ftp server is known throughout
      * the commons-net ftp system.
@@ -185,7 +191,7 @@ public class FTPClientConfig
      *
      * @since 1.5
      */
-    public static final String SYST_L8 = "Type: L8";
+    public static final String SYST_L8 = "TYPE: L8";
 
     /**
      * Identifier by which an Netware-based ftp server is known throughout
diff --git a/src/java/org/apache/commons/net/ftp/parser/DefaultFTPFileEntryParserFactory.java b/src/java/org/apache/commons/net/ftp/parser/DefaultFTPFileEntryParserFactory.java
index 8015f100f..5a5ad1eed 100644
--- a/src/java/org/apache/commons/net/ftp/parser/DefaultFTPFileEntryParserFactory.java
+++ b/src/java/org/apache/commons/net/ftp/parser/DefaultFTPFileEntryParserFactory.java
@@ -28,7 +28,7 @@
  * org.apache.commons.net.ftp.FTPClient.listFiles()
  * if no other implementation has been specified.
  *
- * @see org.apache.commons.net.ftp.FTPClient#listFiles
+ * @see org.apache.commons.net.ftp.FTPClient#listFiles()
  * @see org.apache.commons.net.ftp.FTPClient#setParserFactory
  */
 public class DefaultFTPFileEntryParserFactory
@@ -109,7 +109,8 @@ else if (ukey.indexOf(FTPClientConfig.SYST_OS2) >= 0)
                 {
                     parser = createOS2FTPEntryParser();
                 }
-                else if (ukey.indexOf(FTPClientConfig.SYST_OS400) >= 0)
+                else if ((ukey.indexOf(FTPClientConfig.SYST_OS400) >= 0)
+                    || (ukey.indexOf(FTPClientConfig.SYST_AS400) >= 0))
                 {
                     parser = createOS400FTPEntryParser();
                 }
diff --git a/src/test/org/apache/commons/net/ftp/parser/DefaultFTPFileEntryParserFactoryTest.java b/src/test/org/apache/commons/net/ftp/parser/DefaultFTPFileEntryParserFactoryTest.java
index 6a9aa64af..98b626cab 100644
--- a/src/test/org/apache/commons/net/ftp/parser/DefaultFTPFileEntryParserFactoryTest.java
+++ b/src/test/org/apache/commons/net/ftp/parser/DefaultFTPFileEntryParserFactoryTest.java
@@ -62,6 +62,15 @@ public void testDefaultParserFactory() throws Exception {
         parser = factory.createFileEntryParser("OS/400");
         assertTrue(parser instanceof CompositeFileEntryParser);
 
+        parser = factory.createFileEntryParser("AS/400");
+        assertTrue(parser instanceof CompositeFileEntryParser);
+
+        // Added test to make sure it handles the Unix systems that were
+        // compiled with OS as "UNKNOWN". This test validates that the
+        // check is case-insensitive.
+        parser = factory.createFileEntryParser("UNKNOWN Type: L8");
+
+
         try {
             parser = factory.createFileEntryParser("OS2FTPFileEntryParser");
             fail("Exception should have been thrown. \"OS2FTPFileEntryParser\" is not a recognized key");
diff --git a/xdocs/changes.xml b/xdocs/changes.xml
index bc7a88c13..715a1e718 100644
--- a/xdocs/changes.xml
+++ b/xdocs/changes.xml
@@ -97,6 +97,10 @@ limitations under the License.
             </action>
             <action dev="sebb" type="fix" issue="NET-223">
                 the data connection socket is not closed when an IOException occurred
+            </action>
+            <action dev="sebb" type="fix" issue="NET-231">
+                ParserInitializationException when connecting to a Unix FTP server: comparison string must be upper case
+                Added check for AS/400 response.
             </action>
 		</release> 
 		
